curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/inpaint.html

```diff
@@ -0,0 +1,129 @@
+<html>
+<head>
+<meta charset="utf-8">
+<title>CVdrawing</title>
+</head>
+<body>
+<p id="status">OpenCV.js is loading...</p>
+<canvas id="canvas" width="640" height="426">
+</canvas>
+<img id="src" src="bird.jpg" style="display:none"/>
+<input type="button" id="myButton" value="Test" disabled=true
+onclick="inpaint()"/>
+
+<script type="text/javascript">
+
+var checker = false;
+var src;
+var mask;
+var dst;
+var p_x = 0;
+var p_y = 0;
+var paint;
+var srcmat;
+var maskmat;
+var dstmat;
+var x;
+var y;
+
+window.onload = function()
+{
+context = document.getElementById("canvas").getContext("2d");
+img = document.getElementById("src");
+context.drawImage(img, 0, 0);
+};
+
+document.getElementById("canvas").onmousedown = function(e)
+{
+  paint = true;
+  x = e.pageX - this.offsetLeft;
+  y = e.pageY - this.offsetTop;
+  if (p_x == 0)
+  {
+      p_x = x;
+      p_y = y;
+  }
+  draw();
+};
+
+document.getElementById("canvas").onmousemove = function(e)
+{
+  if(paint)
+  {
+    cv.line(src, new cv.Point(p_x, p_y), new cv.Point(x, y), new cv.Scalar(255, 255, 255, 255), 2);
+    cv.line(maskmat, new cv.Point(p_x, p_y), new cv.Point(x, y), new cv.Scalar(255, 255, 255, 255), 2);
+    p_x = x;
+    p_y = y;
+    x = e.pageX - this.offsetLeft;
+    y = e.pageY - this.offsetTop;
+    draw();
+  }
+};
+
+document.getElementById("canvas").onmouseup = function(e)
+{
+  paint = false;
+  p_x = 0;
+  p_y = 0;
+};
+
+document.getElementById("canvas").onmouseleave = function(e)
+{
+  paint = false;
+};
+
+function init()
+{
+  if (checker == true)
+  {
+    src = cv.imread("src");;
+    dst = src.clone();
+    let cols = src.size().width;
+    let rows = src.size().height;
+    let ch = src.channels();
+    srcmat = new cv.Mat();
+    maskmat = new cv.Mat(426, 640, cv.CV_8UC1, new cv.Scalar(0, 0, 0, 255));;
+    dstmat = new cv.Mat();
+  }
+}
+
+function draw()
+{
+
+    if (paint == true)
+    {
+        cv.imshow("canvas", src);
+    }
+}
```

Comment:
It seems to me that we need to call this method only if image is changed (at the onmousemove event). So I think we can just call imshow there.

Reply:
We need it for drawing current point on mousedown event.

Reply:
Ok, I see but is it works now? `cv.line` only at mousemove

Reply:
I test sample, it is works correctly.

Reply:
If we remove `draw()` from `onmousedown` it works as expected?

Reply:
Indeed, if we remove draw() completely (from onmousedown and onmousemove) and add cv.imshow() to onmousemove, sample working correctly.

Reply:
Can we resolve current conversation?

Reply:
Sure